src/test/java/org/mule/extensions/jms/test/connection/JmsReconnectionTestCase.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.jms.test.connection;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.junit.rules.ExpectedException.none;
+import static org.junit.rules.RuleChain.outerRule;
+import static org.mule.extensions.jms.test.AllureConstants.JmsFeature.JMS_EXTENSION;
+import static org.mule.extensions.jms.test.AllureConstants.JmsFeature.JmsStory.RECONNECTION;
+import static org.mule.extensions.jms.test.JmsMessageStorage.pollMuleMessage;
+
+import org.mule.extensions.jms.api.exception.JmsConsumeException;
+import org.mule.extensions.jms.test.JmsAbstractTestCase;
+import org.mule.extensions.jms.test.JmsMessageStorage;
+import org.mule.extensions.jms.test.util.ActiveMQBroker;
+import org.mule.runtime.api.message.Message;
+import org.mule.runtime.core.api.construct.Flow;
+import org.mule.runtime.core.api.exception.MessagingException;
+import org.mule.tck.junit4.rule.SystemProperty;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Step;
+import io.qameta.allure.Story;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+
+@Feature(JMS_EXTENSION)
+@Story(RECONNECTION)
+public class JmsReconnectionTestCase extends JmsAbstractTestCase {
+
+  public static final String RECONNECTED_LISTENER_FLOW = "reconnected-listener";
+
+  public static final String RECONNECTION_FAIL_LISTENER_FLOW = "fail-reconnect-listener";
+
+  private static ActiveMQBroker amqBroker = new ActiveMQBroker("amqPort") {
+
+    @Override
+    protected void before() throws Throwable {
+      amqBroker.start();
+    }
+
+    @Override
+    protected void after() {
+      amqBroker.stop();
+    }
+  };
+
+  private static SystemProperty amqUrl = new SystemProperty("amq.url", amqBroker.getConnectorUrl());
+
+  @ClassRule
+  public static TestRule chain = outerRule(amqBroker).around(amqUrl);
+
+  private final String listenerDest = newDestination("listenerDest");
+  private final String failedListenerDest = newDestination("failedListenerDest");
+  private final String publishDest = newDestination("publishDest");
+
+  @Rule
```

Comment:
Rule, as opposed to ClassRule, will reset these properties for every test method in this class. Is this necessary?

Reply:
Good catch. Thanks

Reply:
Nevertheless, that change will imply a considerable refactor since the @ClassRule requires the members to be static those destinations are created with a method if the ancestor class which is not. 

Do you think, this improvement worth doing the refactor? I'm not that sure.

Reply:
you always need to do: `new SystemProperty("listenerDest", newDestination("listenerDest"));` to ensure that between test cases you don't use a queue with a dirty state